items:
- uid: System.ObjectDisposedException
  commentId: T:System.ObjectDisposedException
  id: ObjectDisposedException
  parent: System
  children:
  - System.ObjectDisposedException.#ctor(System.String)
  - System.ObjectDisposedException.#ctor(System.String,System.Exception)
  - System.ObjectDisposedException.#ctor(System.String,System.String)
  - System.ObjectDisposedException.Message
  - System.ObjectDisposedException.ObjectName
  langs:
  - csharp
  - vb
  name: ObjectDisposedException
  fullName: System.ObjectDisposedException
  type: Class
  source:
    remote:
      path: src/System.Runtime/ref/System.Runtime.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: ObjectDisposedException
    path: src/System.Runtime/ref/System.Runtime.cs
    startLine: 1320
  assemblies:
  - System.Runtime
  namespace: System
  summary: "\n<p>The exception that is thrown when an operation is performed on a disposed object.</p>\n"
  remarks: "\n<p>An <xref href=\"System.ObjectDisposedException\" data-throw-if-not-resolved=\"false\"></xref> is thrown when you try to access a member of an object that implements the <xref href=\"System.IDisposable\" data-throw-if-not-resolved=\"false\"></xref> interface, and that object has been disposed. Typically, this exception is caused by one of the following conditions: </p>\n<ul><li>\n<p>You've called an object's <strong>Dispose</strong> method, and you’re trying to access an instance member that gets or sets the object's state. The following example illustrates the <xref href=\"System.ObjectDisposedException\" data-throw-if-not-resolved=\"false\"></xref> that is thrown when you try to reset the frequency of timer notifications after you call the <xref href=\"System.Threading.Timer.Dispose\" data-throw-if-not-resolved=\"false\"></xref> method. </p>System.ObjectDisposedException#1</li><li>\n<p>You've called an object's <strong>Close</strong> method, and you’re trying to access an instance member that gets or sets the object's state. Often, the <strong>Close</strong> method provides a type's public implementation of the <xref href=\"System.IDisposable.Dispose\" data-throw-if-not-resolved=\"false\"></xref> method. </p>\n</li><li>\n<p>You've called an object's <strong>Dispose</strong> method multiple times. Typically, this doesn’t throw an exception. However, depending on how a type implements <xref href=\"System.IDisposable.Dispose\" data-throw-if-not-resolved=\"false\"></xref>, it may not allow multiple calls to <strong>Dispose</strong>.  </p>\n</li></ul>\n<p>In most cases, this exception results from developer error. Instead of handling the error in a <strong>try</strong>/<strong>catch</strong> block, you should correct the error, typically by reinstantiating the object. </p>\n"
  example:
  - "\n<p>The following example demonstrates an error that causes the <strong>ObjectDisposedException</strong> exception to be thrown.</p>ObjDispEx#1<p>This code produces the following output: </p><pre><code>Caught:\nCannot access a closed Stream.</code></pre>"
  syntax:
    content: 'public class ObjectDisposedException : InvalidOperationException'
    content.vb: >-
      Public Class ObjectDisposedException
          Inherits InvalidOperationException
  seealso:
  - type: System.IDisposable
    commentId: T:System.IDisposable
  inheritance:
  - System.Object
  - System.Exception
  - System.InvalidOperationException
  inheritedMembers:
  - System.Exception.Data
  - System.Exception.HelpLink
  - System.Exception.HResult
  - System.Exception.InnerException
  - System.Exception.Source
  - System.Exception.StackTrace
  - System.Exception.GetBaseException
  - System.Exception.ToString
  - System.Object.Equals(System.Object)
  - System.Object.Equals(System.Object,System.Object)
  - System.Object.GetHashCode
  - System.Object.GetType
  - System.Object.MemberwiseClone
  - System.Object.ReferenceEquals(System.Object,System.Object)
  modifiers.csharp:
  - public
  - class
  modifiers.vb:
  - Public
  - Class
- uid: System.ObjectDisposedException.#ctor(System.String)
  commentId: M:System.ObjectDisposedException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.ObjectDisposedException
  langs:
  - csharp
  - vb
  name: ObjectDisposedException(String)
  fullName: System.ObjectDisposedException.ObjectDisposedException(System.String)
  type: Constructor
  source:
    remote:
      path: src/System.Runtime/ref/System.Runtime.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.Runtime/ref/System.Runtime.cs
    startLine: 1322
  assemblies:
  - System.Runtime
  namespace: System
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.ObjectDisposedException\" data-throw-if-not-resolved=\"false\"></xref> class with a string containing the name of the disposed object.</p>\n"
  remarks: "\n<p>The <xref href=\"System.ObjectDisposedException.Message\" data-throw-if-not-resolved=\"false\"></xref> property is initialized to a system-supplied message that describes the error and includes the <em>objectName</em> parameter. This message takes into account the current system culture.</p>\n<p>The <strong>InnerException</strong> property of the new instance is initialized to <strong>null</strong>.</p>\n<p>If <em>objectName</em> is <strong>null</strong>, the <strong>Message</strong> property contains only an error message.</p>\n"
  syntax:
    content: public ObjectDisposedException(string objectName)
    content.vb: Public Sub New(objectName As String)
    parameters:
    - id: objectName
      type: System.String
      description: "\n<p>A string containing the name of the disposed object. </p>\n"
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
- uid: System.ObjectDisposedException.#ctor(System.String,System.Exception)
  commentId: M:System.ObjectDisposedException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.ObjectDisposedException
  langs:
  - csharp
  - vb
  name: ObjectDisposedException(String, Exception)
  fullName: System.ObjectDisposedException.ObjectDisposedException(System.String, System.Exception)
  type: Constructor
  source:
    remote:
      path: src/System.Runtime/ref/System.Runtime.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.Runtime/ref/System.Runtime.cs
    startLine: 1323
  assemblies:
  - System.Runtime
  namespace: System
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.ObjectDisposedException\" data-throw-if-not-resolved=\"false\"></xref> class with a specified error message and a reference to the inner exception that is the cause of this exception.</p>\n"
  remarks: "\n<p>An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property. The <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property returns the same value that is passed into the constructor.</p>\n"
  syntax:
    content: public ObjectDisposedException(string message, Exception innerException)
    content.vb: Public Sub New(message As String, innerException As Exception)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>The error message that explains the reason for the exception.</p>\n"
    - id: innerException
      type: System.Exception
      description: "\n<p>The exception that is the cause of the current exception. If <em>innerException</em> is not <strong>null</strong>, the current exception is raised in a <strong>catch</strong> block that handles the inner exception.</p>\n"
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
- uid: System.ObjectDisposedException.#ctor(System.String,System.String)
  commentId: M:System.ObjectDisposedException.#ctor(System.String,System.String)
  id: '#ctor(System.String,System.String)'
  parent: System.ObjectDisposedException
  langs:
  - csharp
  - vb
  name: ObjectDisposedException(String, String)
  fullName: System.ObjectDisposedException.ObjectDisposedException(System.String, System.String)
  type: Constructor
  source:
    remote:
      path: src/System.Runtime/ref/System.Runtime.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.Runtime/ref/System.Runtime.cs
    startLine: 1324
  assemblies:
  - System.Runtime
  namespace: System
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.ObjectDisposedException\" data-throw-if-not-resolved=\"false\"></xref> class with the specified object name and message.</p>\n"
  remarks: "\n<p>The <strong>InnerException</strong> property of the new instance is initialized to <strong>null</strong>.</p>\n<p>If <em>objectName</em> is <strong>null</strong>, the <strong>Message</strong> property contains only an error message.</p>\n"
  syntax:
    content: public ObjectDisposedException(string objectName, string message)
    content.vb: Public Sub New(objectName As String, message As String)
    parameters:
    - id: objectName
      type: System.String
      description: "\n<p>The name of the disposed object. </p>\n"
    - id: message
      type: System.String
      description: "\n<p>The error message that explains the reason for the exception. </p>\n"
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
- uid: System.ObjectDisposedException.Message
  commentId: P:System.ObjectDisposedException.Message
  id: Message
  parent: System.ObjectDisposedException
  langs:
  - csharp
  - vb
  name: Message
  fullName: System.ObjectDisposedException.Message
  type: Property
  source:
    remote:
      path: src/System.Runtime/ref/System.Runtime.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: Message
    path: src/System.Runtime/ref/System.Runtime.cs
    startLine: 1325
  assemblies:
  - System.Runtime
  namespace: System
  summary: "\n<p>Gets the message that describes the error.</p>\n"
  remarks: "\n<p>If the <xref href=\"System.ObjectDisposedException.ObjectName\" data-throw-if-not-resolved=\"false\"></xref> property is not <strong>null</strong>, the message includes the name of the object.</p>\n<p>This property overrides <xref href=\"System.Exception.Message\" data-throw-if-not-resolved=\"false\"></xref>.</p>\n"
  syntax:
    content: public override string Message { get; }
    content.vb: Public Overrides ReadOnly Property Message As String
    parameters: []
    return:
      type: System.String
      description: "\n<p>A string that describes the error.</p>\n"
  overridden: System.Exception.Message
  modifiers.csharp:
  - public
  - override
  - get
  modifiers.vb:
  - Public
  - Overrides
  - ReadOnly
- uid: System.ObjectDisposedException.ObjectName
  commentId: P:System.ObjectDisposedException.ObjectName
  id: ObjectName
  parent: System.ObjectDisposedException
  langs:
  - csharp
  - vb
  name: ObjectName
  fullName: System.ObjectDisposedException.ObjectName
  type: Property
  source:
    remote:
      path: src/System.Runtime/ref/System.Runtime.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: ObjectName
    path: src/System.Runtime/ref/System.Runtime.cs
    startLine: 1326
  assemblies:
  - System.Runtime
  namespace: System
  summary: "\n<p>Gets the name of the disposed object.</p>\n"
  remarks: "\n<p>If the current property is not <strong>null</strong> or <xref href=\"System.String.Empty\" data-throw-if-not-resolved=\"false\"></xref>, the value of this property is included in the string returned by the <xref href=\"System.ObjectDisposedException.Message\" data-throw-if-not-resolved=\"false\"></xref> property.</p>\n"
  syntax:
    content: public string ObjectName { get; }
    content.vb: Public ReadOnly Property ObjectName As String
    parameters: []
    return:
      type: System.String
      description: "\n<p>A string containing the name of the disposed object.</p>\n"
  modifiers.csharp:
  - public
  - get
  modifiers.vb:
  - Public
  - ReadOnly
references:
- uid: System
  isExternal: false
  name: System
  fullName: System
- uid: System.Object
  parent: System
  isExternal: false
  name: Object
  fullName: System.Object
- uid: System.Exception
  parent: System
  isExternal: false
  name: Exception
  fullName: System.Exception
- uid: System.InvalidOperationException
  parent: System
  isExternal: false
  name: InvalidOperationException
  fullName: System.InvalidOperationException
- uid: System.Exception.Data
  parent: System.Exception
  isExternal: false
  name: Data
  fullName: System.Exception.Data
- uid: System.Exception.HelpLink
  parent: System.Exception
  isExternal: false
  name: HelpLink
  fullName: System.Exception.HelpLink
- uid: System.Exception.HResult
  parent: System.Exception
  isExternal: false
  name: HResult
  fullName: System.Exception.HResult
- uid: System.Exception.InnerException
  parent: System.Exception
  isExternal: false
  name: InnerException
  fullName: System.Exception.InnerException
- uid: System.Exception.Source
  parent: System.Exception
  isExternal: false
  name: Source
  fullName: System.Exception.Source
- uid: System.Exception.StackTrace
  parent: System.Exception
  isExternal: false
  name: StackTrace
  fullName: System.Exception.StackTrace
- uid: System.Exception.GetBaseException
  parent: System.Exception
  isExternal: false
  name: GetBaseException()
  fullName: System.Exception.GetBaseException()
  spec.csharp:
  - uid: System.Exception.GetBaseException
    name: GetBaseException
    fullName: System.Exception.GetBaseException
  - name: (
    fullName: (
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Exception.GetBaseException
    name: GetBaseException
    fullName: System.Exception.GetBaseException
  - name: (
    fullName: (
  - name: )
    fullName: )
- uid: System.Exception.ToString
  parent: System.Exception
  isExternal: false
  name: ToString()
  fullName: System.Exception.ToString()
  spec.csharp:
  - uid: System.Exception.ToString
    name: ToString
    fullName: System.Exception.ToString
  - name: (
    fullName: (
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Exception.ToString
    name: ToString
    fullName: System.Exception.ToString
  - name: (
    fullName: (
  - name: )
    fullName: )
- uid: System.Object.Equals(System.Object)
  parent: System.Object
  isExternal: false
  name: Equals(Object)
  fullName: System.Object.Equals(System.Object)
  spec.csharp:
  - uid: System.Object.Equals(System.Object)
    name: Equals
    fullName: System.Object.Equals
  - name: (
    fullName: (
  - uid: System.Object
    name: Object
    fullName: System.Object
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Object.Equals(System.Object)
    name: Equals
    fullName: System.Object.Equals
  - name: (
    fullName: (
  - uid: System.Object
    name: Object
    fullName: System.Object
  - name: )
    fullName: )
- uid: System.Object.Equals(System.Object,System.Object)
  parent: System.Object
  isExternal: false
  name: Equals(Object, Object)
  fullName: System.Object.Equals(System.Object, System.Object)
  spec.csharp:
  - uid: System.Object.Equals(System.Object,System.Object)
    name: Equals
    fullName: System.Object.Equals
  - name: (
    fullName: (
  - uid: System.Object
    name: Object
    fullName: System.Object
  - name: ', '
    fullName: ', '
  - uid: System.Object
    name: Object
    fullName: System.Object
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Object.Equals(System.Object,System.Object)
    name: Equals
    fullName: System.Object.Equals
  - name: (
    fullName: (
  - uid: System.Object
    name: Object
    fullName: System.Object
  - name: ', '
    fullName: ', '
  - uid: System.Object
    name: Object
    fullName: System.Object
  - name: )
    fullName: )
- uid: System.Object.GetHashCode
  parent: System.Object
  isExternal: false
  name: GetHashCode()
  fullName: System.Object.GetHashCode()
  spec.csharp:
  - uid: System.Object.GetHashCode
    name: GetHashCode
    fullName: System.Object.GetHashCode
  - name: (
    fullName: (
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Object.GetHashCode
    name: GetHashCode
    fullName: System.Object.GetHashCode
  - name: (
    fullName: (
  - name: )
    fullName: )
- uid: System.Object.GetType
  parent: System.Object
  isExternal: false
  name: GetType()
  fullName: System.Object.GetType()
  spec.csharp:
  - uid: System.Object.GetType
    name: GetType
    fullName: System.Object.GetType
  - name: (
    fullName: (
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Object.GetType
    name: GetType
    fullName: System.Object.GetType
  - name: (
    fullName: (
  - name: )
    fullName: )
- uid: System.Object.MemberwiseClone
  parent: System.Object
  isExternal: false
  name: MemberwiseClone()
  fullName: System.Object.MemberwiseClone()
  spec.csharp:
  - uid: System.Object.MemberwiseClone
    name: MemberwiseClone
    fullName: System.Object.MemberwiseClone
  - name: (
    fullName: (
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Object.MemberwiseClone
    name: MemberwiseClone
    fullName: System.Object.MemberwiseClone
  - name: (
    fullName: (
  - name: )
    fullName: )
- uid: System.Object.ReferenceEquals(System.Object,System.Object)
  parent: System.Object
  isExternal: false
  name: ReferenceEquals(Object, Object)
  fullName: System.Object.ReferenceEquals(System.Object, System.Object)
  spec.csharp:
  - uid: System.Object.ReferenceEquals(System.Object,System.Object)
    name: ReferenceEquals
    fullName: System.Object.ReferenceEquals
  - name: (
    fullName: (
  - uid: System.Object
    name: Object
    fullName: System.Object
  - name: ', '
    fullName: ', '
  - uid: System.Object
    name: Object
    fullName: System.Object
  - name: )
    fullName: )
  spec.vb:
  - uid: System.Object.ReferenceEquals(System.Object,System.Object)
    name: ReferenceEquals
    fullName: System.Object.ReferenceEquals
  - name: (
    fullName: (
  - uid: System.Object
    name: Object
    fullName: System.Object
  - name: ', '
    fullName: ', '
  - uid: System.Object
    name: Object
    fullName: System.Object
  - name: )
    fullName: )
- uid: System.String
  parent: System
  isExternal: false
  name: String
  fullName: System.String
- uid: System.Exception.Message
  parent: System.Exception
  isExternal: false
  name: Message
  fullName: System.Exception.Message
